### PR TITLE
Homework 1

### PR DESCRIPTION
**Pull request by potato1543**

### CODE REVIEW

File: cs4660/graph/graph.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        for edge in self.adjacency_list[node]:
+          rest.append(edge.to_node)
+        return rest
+        
+
+    def add_node(self, node):
+        if node in self.adjacency_list:
+           return False
+        else:
+            self.adjacency_list[node] = {}
```
Comment (rcliao) [inline cs4660/graph/graph.py]: data structure used here for the value of the dictionary is wrong. You are assigning the value to be another dictionary. (which is why the code fails when you call `append` method down at `add_edge` method)

File: cs4660/graph/graph.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    AdjacencyList is one of the graph representation which uses adjacency list to
+    store nodes and edges
+    """
+    def __init__(self):
+        # adjacencyList should be a dictonary of node to edges
+        self.adjacency_list = {}
+
+    def adjacent(self, node_1, node_2):
+        if node_1 in self.adjacency_list:
+            if node_2 in self.adjacency_list[node_1]:
```
Comment (rcliao) [inline cs4660/graph/graph.py]: what data structure do you have here? A dictionary of node to node? If so, the structure is wrong; the dictionary should contain key of node and value of list of either edge or tuples.

File: cs4660/graph/graph.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+                 return True
+        return False
+
+    def neighbors(self, node):
+        rest = []
+        for edge in self.adjacency_list[node]:
+          rest.append(edge.to_node)
```
Comment (rcliao) [inline cs4660/graph/graph.py]: Indentation is inconsistent. Did you just copy this line of code from somewhere else?

File: cs4660/graph/graph.py [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+        adj = self.adjacency_list[edge.from_node]     
+        adj.append(edge.to_node)
+        return True
+
+    def remove_edge(self, edge):
+         if edge.from_node in self.adjacency_list:
+             if edge.to_node in self.adjacency_list[edge.from_node]:
+                 self.adjacency_list[edge.from_node].remove(edge.to_node)
+                 return True
+         return False
```
Comment (rcliao) [inline cs4660/graph/graph.py]: It seems like your code here are very inconsistent on the data structure used for the `adjacency_list`. Some method assumes the adjacency_list to be a dictionary of node to a list of nodes, some assumes the structure to be a dictionary of node to a list of edges while some assumes the dictionary to be node to a dictionary of nodes.

File: cs4660/graph/graph.py [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+         else:
+             return True
+  
+
+    def neighbors(self, node):
+      if node not in self.nodes:
+             return [Node(neighbor_number)
+                 for neighbor_number
+                    in range(0, len(self.adjacency_matrix[self.__get_node_index(node)]))
+                if self.adjacency_matrix[self.__get_node_index(node)][neighbor_number] is not None]
```
Comment (rcliao) [inline cs4660/graph/graph.py]: Indentation?

File: cs4660/graph/graph.py [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+
+    def add_node(self, node):
+        if node in self.nodes:
+             return False
+        self.nodes.append(node)
+        for i in range(len(self.adjacency_matrix)):
+             self.adjacency_matrix[i].append(0)
```
Comment (rcliao) [inline cs4660/graph/graph.py:193]: Your code append `0` yet you check for `None` above in the adjacent method?

**Comment by potato1543** [comment]
I was just trying different methods. I am going to continue to work on it. With the feedback thank you

From Kaila Mayho

**Pull request merged**